anomalib/data/mvtec_loco.py

```diff
@@ -0,0 +1,863 @@
+            if mask_paths is None:
+                return {
+                    "masks": None,
+                    "mask": None,
+                }
+
+            # iterate the mask paths and read the masks returning a tuple of masks
+            masks: Tuple[np.ndarray, ...] = tuple(
+                _binarize_mask_float(read_mask(mask_path)) for mask_path in mask_paths
+            )
+
+            return {
+                # replace the tuple of masks by a single array where each anomaly has
+                # a different value encoding an individual anomaly region
+                "masks": self._sum_masks(masks),
+                # combine the multiple masks into a single (binary) mask
+                "mask": np.stack(masks, axis=0).sum(axis=0).clip(0, 1),
+            }
+
+        raise NotImplementedError(f"Imread strategy '{self.imread_strategy}' is not supported.")
+
+    def __getitem__(self, index: int) -> Union[Dict[str, Tensor], Dict[str, Union[str, Tensor, int]]]:
+        """Get dataset item for the index ``index``.
+
+        Args:
+            index (int): Index to get the item.
+
+        Returns:
+            Union[Dict[str, Tensor], Dict[str, Union[str, Tensor]]]: Dict of image tensor during training.
+                Otherwise, Dict containing image path, image tensor, label, anomaly type and,
+                if it is segmentation task, mask path and mask tensor.
+        """
+        item: Dict[str, Union[str, Tensor, int]] = {}
+
+        image = self._get_image(index)
+        pre_processed = self.pre_process(image=image)
+        item = {
+            "image": pre_processed["image"],
+        }
+
+        if self.split not in (SPLIT_VALIDATION, SPLIT_TEST):
+            return item
+
+        item.update(
+            {
+                "label": self.samples.iloc[index]["label"],
+                "image_path": str(self.samples.iloc[index]["image_path"]),
+                "anotype": self.samples.iloc[index]["anotype"],
+                "super_anotype": self.samples.iloc[index]["super_anotype"],
+            }
+        )
+
+        if self.task != TASK_SEGMENTATION:
+            return item
+
+        mask_dict: Dict[str, ndarray]
+
+        # Only Anomalous (1) images has masks in MVTec LOCO AD dataset.
+        # Therefore, create empty mask for Normal (0) images.
+        if self.samples.iloc[index]["label"] == LABEL_NORMAL:
+            mask = np.zeros(shape=image.shape[:2])  # shape: (H, W, C)
+            mask_dict = {"mask": mask, "masks": mask}
+
+        else:
+            mask_dict = self._get_masks(index)
+
+        item.update(
+            {
+                "mask_paths": str(self.samples.iloc[index]["mask_paths"]),
+                # TODO CHECK IF THE DOUBLE CALL TO PREPROCESS WILL WORK WITH ALBUMENTATIONS
+                "masks": self.pre_process(image=image, mask=mask_dict["masks"])["mask"],
+                "mask": self.pre_process(image=image, mask=mask_dict["mask"])["mask"],
```

Comment:
self.pre_process is being called for the 3rd time here, will that create any problems?

I'm thinking that maybe the random transforms will apply the same transform every two times (for the image and for the mask).